tests/e2e/idempotency/test_idempotency_dynamodb.py

```diff
@@ -97,3 +102,30 @@ def test_parallel_execution_idempotency(parallel_execution_handler_fn_arn: str):
     # THEN
     assert "Execution already in progress with idempotency key" in error_idempotency_execution_response
     assert "Task timed out after" in timeout_execution_response
+
+
+@pytest.mark.xdist_group(name="idempotency")
+def test_parallel_functions_execution_idempotency(parallel_functions_handler_fn_arn: str):
```

Comment:
I think this name might be confusing for someone looking at the code. The same goes for the `test_parallel_execution_idempotency` test. We should take this opportunity to rename both. Let me think of good names for this or you can suggest.

Reply:
happy to hear your suggestion, I will try to come up with better proposal this evening also

Reply:
@leandrodamascena Renamed the test_parallel_functions_execution_idempotency to test_idempotent_function_thread_safety ( + renamed handlers accordingly). Let me know what do you think. Have no idea how to rename test_parallel_execution_idempotency so any suggestion welcomed :-)

Reply:
Thank you, @mploski! Everything is fine now. With this change, clean up what you're testing and avoid confusion.

I don't have a good name for the other function right now, so we can merge this PR and I'll add in my backlog to rename this function in next releases.